services/httpservice/client.go

```diff
@@ -30,6 +30,38 @@ func IsReservedIP(ip net.IP) bool {
 	return false
 }
 
+func IsOwnIP(ip net.IP) (bool, error) {
+	interfaces, err := net.Interfaces()
+
```

Comment:
style nit: spurious blank line between call and error check, here and below.

Reply:
:heavy_check_mark: